### PR TITLE
Add docs on accessing Azure blob storage through fsspec

### PR DESCRIPTION
**Pull request by sgrigory**
### Changes

Adding an example of DataPipe usage with Azure Blob storage via `fsspec`, similar to https://github.com/pytorch/data/pull/812. The example is placed into a new section in `docs/source/tutorial.rst`

Here is the screenshot showing that code snippets in the tutorial work as expected:

Screenshot 2022-10-18 at 19 33 49

####  Minor note

Technically, `fsspec` [allows both path prefixes `abfs://` or `az://`](https://github.com/fsspec/adlfs/blob/f15c37a43afd87a04f01b61cd90294dd57181e1d/README.md?plain=1#L33) for Azure Blob storage Gen2 as synonyms. However, only `abfs://` works for us for the following reason:
- If a path starts with `az`, the variable `fs.protocol` [here](https://github.com/pytorch/data/blob/768ecdae8b56af640a78e29f82864dc4f65df371/torchdata/datapipes/iter/load/fsspec.py#L82) is still `abfs`
- So the condition `root.startswith(protocol)` is false, and `is_local` is true
- As a result the path "doubles" in [this line](https://github.com/pytorch/data/blob/768ecdae8b56af640a78e29f82864dc4f65df371/torchdata/datapipes/iter/load/fsspec.py#L95), like on this screenshot:
Screenshot 2022-10-18 at 19 50 56

This won't have any effect for the users, however, as long as they use `abfs://` prefix recommended in the tutorial

### CODE REVIEW

**Review by ejguan — APPROVED** [review]
This is great! Thank you. And, do you mind opening an issue about your minor note? I think it's a bug.

And, adding original author @NivekT of this tutorial to review.

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -298,7 +298,7 @@ recommend using the functional form of DataPipes.
 Working with Cloud Storage Providers
 ---------------------------------------------
 
-In this section, we show examples accessing AWS S3 and Google Cloud Storage with built-in``fsspec`` DataPipes.
+In this section, we show examples accessing AWS S3 and Google Cloud Storage with built-in ``fsspec`` DataPipes.
```
Comment (NivekT) [inline docs/source/tutorial.rst:301]:
```suggestion
In this section, we show examples accessing AWS S3, Google Cloud Storage, and Azure Cloud Storage with built-in ``fsspec`` DataPipes.
```

**Review by NivekT — APPROVED** [review]
One nit change. LGTM! Thank you so much for submitting this!

cc: @msaroufim

**Comment by msaroufim** [comment]
Thank you for the speed @sgrigory! Let me know if you want more stuff to do!

**Review by msaroufim — APPROVED** [review]

**Comment by ejguan** [comment]
@sgrigory Ops, I thought this PR came from OSS. Feel free to take over the imported Diff and land it. I don't want to steal credit from your valuable work! Thanks!

**Comment by sgrigory** [comment]
> This is great! Thank you. And, do you mind opening an issue about your minor note? I think it's a bug.
> 
> And, adding original author @NivekT of this tutorial to review.

Thanks for the review - opened an issue #840

**Pull request closed without merging**